pandas/core/reshape/reshape.py

```diff
@@ -979,22 +981,25 @@ def wide_to_long(df, stubnames, i, j, sep="", suffix='\d+'):
 
     Less wieldy column names are also handled
 
+    >>> np.random.seed(0)
     >>> df = pd.DataFrame({'A(quarterly)-2010': np.random.rand(3),
     ...                    'A(quarterly)-2011': np.random.rand(3),
     ...                    'B(quarterly)-2010': np.random.rand(3),
     ...                    'B(quarterly)-2011': np.random.rand(3),
     ...                    'X' : np.random.randint(3, size=3)})
     >>> df['id'] = df.index
     >>> df
-      A(quarterly)-2010 A(quarterly)-2011 B(quarterly)-2010 B(quarterly)-2011
-    0          0.531828          0.724455          0.322959          0.293714
-    1          0.634401          0.611024          0.361789          0.630976
-    2          0.849432          0.722443          0.228263          0.092105
-    \
+    ... # doctest: +NORMALIZE_WHITESPACE
+       A(quarterly)-2010  A(quarterly)-2011  B(quarterly)-2010  B(quarterly)-2011  \
+    0           0.548814           0.544883           0.437587           0.383442
+    1           0.715189           0.423655           0.891773           0.791725
+    2           0.602763           0.645894           0.963663           0.528895
+    
        X  id
     0  0   0
     1  1   1
-    2  2   2
+    2  1   2
+    
     >>> pd.wide_to_long(df, ['A(quarterly)', 'B(quarterly)'],
                         i='id', j='year', sep='-')
              X     A(quarterly)  B(quarterly)
```

Comment:
X is random ... which didn't make sense to me.
`    >>> np.random.seed(0)
    >>> df = pd.DataFrame({'A(quarterly)-2010': np.random.rand(3),
    ...                    'A(quarterly)-2011': np.random.rand(3),
    ...                    'B(quarterly)-2010': np.random.rand(3),
    ...                    'B(quarterly)-2011': np.random.rand(3),
    ...                    'X' : np.random.randint(3, size=3)})
    >>> df['id'] = df.index
    >>> df
    ... # doctest: +NORMALIZE_WHITESPACE
       A(quarterly)-2010  A(quarterly)-2011  B(quarterly)-2010  B(quarterly)-2011  \
    0           0.548814           0.544883           0.437587           0.383442
    1           0.715189           0.423655           0.891773           0.791725
    2           0.602763           0.645894           0.963663           0.528895
       X  id
    0  0   0
    1  1   1
    2  1   2`